### ISSUE TITLE
Update Dex Base Image

### ISSUE THREAD

**Issue by vasartori** [dexidp/dex#1252]
Dex container use alpine:3.4. This image has two (at this moment) High security issues fixable.

**Comment by vasartori** [comment]
@ericchiang Eric, could you please help us with this security improvement?

**Comment by srenatus** [comment]
For what it's worth, it seems that rebuilding v2.10.0 with `alpine:3.4` has resolved those outdated packages issues -- [looking at the Quay security scan](https://quay.io/repository/dexidp/dex?tab=tags), there's no issues for v2.10.0 anymore. I guess the [`apk add` run](https://github.com/dexidp/dex/blob/bb75dcd79354271cdc035e4eedb2455044d6883b/Dockerfile#L14) in the build I've kicked off today has pulled in the more recent packages:

```
$ docker run -it --entrypoint sh quay.io/dexidp/dex:v2.10.0
/ # apk info -v
musl-1.1.14-r16
busybox-1.24.2-r14
alpine-baselayout-3.0.3-r0
alpine-keys-1.1-r0
zlib-1.2.11-r0
libcrypto1.0-1.0.2n-r0
libssl1.0-1.0.2n-r0
apk-tools-2.6.9-r0
scanelf-1.1.6-r0
musl-utils-1.1.14-r16
libc-utils-0.7-r0
ca-certificates-20161130-r0
openssl-1.0.2n-r0
/ #
```

Either way, we've got a release underway #1284, and I'd still favour merging #1253 (perhaps with the alpine version bumped once more?). Thanks for raising this issue! 😃
